### ISSUE TITLE
SC2086 and POSIX set -- $var

### ISSUE THREAD

**Issue by leagris** [koalaman/shellcheck#1763]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh
set -o noglob
read -r
# Explicitly intended word splitting
set -- $REPLY
for arg in "$@"; do
  echo "$arg"
done
```

#### Here's what shellcheck currently says:
```
Line 5	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

In POSIX shell, SC2086 shall ignore unquoted variables in `set -- $varname` because it is the only way to explicitly split words to iterate as the arguments array without forking sub-shell to use external tools like `xargs` to convert a string into arguments.

**Comment by gromgit** [comment]
This is exactly what the [`disable` directive](https://github.com/koalaman/shellcheck/wiki/Directive#disable) is for:
```
# Explicitly intended word splitting
# shellcheck disable=SC2086
set -- $REPLY
```
Disabling SC2086 by default, even if only for POSIX shell, and even if only for `set -- XXX`, means it'll  never catch the cases when it actually **is** a mistake.

**Comment by jasonkarns** [comment]
If an author is using `set -- XXX`, then there is virtually zero chance they _don't_ want word splitting. That is an exceeding common pattern meant specifically to leverage splitting.

I agree that SC2086 should be ignored for this conventional pattern.

Related to #1149 

Both `set --` and `IFS=` are two widely-used patterns that explicitly indicate intentional word splitting. Adding the disable directive for these cases should be considered redundant.

**Comment by forthrin** [comment]
```bash
#!/bin/bash

f="foo bar"

set -- $f # Double quote to prevent globbing and word splitting. [SC2086]
echo "$1" "$2"

read -r -a arr <<< "$f"
set -- "${arr[@]}"
echo "$1" "$2"
```

Workaround above. But in what case would the first method fail or create ambiguity? Preventing word splitting is for preserving spaces, but here one explicitly wants the opposite, to precisely split by them.
